app/lib/utils/oidc_provider.rb

```diff
@@ -0,0 +1,86 @@
+class OidcProvider
+  attr_reader :client_id
+  attr_reader :client_secret
+  attr_reader :provider
+
+  def configure
+    provider = oidc_provider.logical.read("identity/oidc/provider/astral")
+    if provider.nil?
+      create_provider_webapp
+      create_provider_with_email_scope
+      create_entity_for_initial_user
+      create_userpass_for_initial_user
+      map_userpass_to_entity
+    else
+      get_client_info
+    end
+  end
+
+
+  def get_client_info
+    app = oidc_provider.logical.read(WEBAPP_NAME)
+    @client_id = app.data[:client_id]
+    @client_secret = app.data[:client_secret]
+    [ @client_id, @client_secret ]
+  end
+
+  def get_info
+    oidc_provider.logical.read("identity/oidc/provider/astral")
+  end
+
+  private
+  WEBAPP_NAME = "identity/oidc/client/astral"
+
+  def oidc_provider
```

Comment:
i think naming this `vault_client` might make downstream usage a little clearer? since the class is OidcProvider (abstraction) but rubber is meeting road here (vault specifically is the provider).

Reply:
done